### PR TITLE
Initial take

### PR DESCRIPTION
**Pull request by goldmann**

### CODE REVIEW

File: deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkiverse.tekton.client</groupId>
+        <artifactId>quarkus-tekton-client-parent</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>quarkus-tekton-client-deployment</artifactId>
```
Comment (gsmet) [inline deployment/pom.xml:11]: This needs to be `quarkiverse-tekton-client...` Same for the other artifacts.

File: deployment/pom.xml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    <name>Quarkus - Tekton Client - Deployment</name>
```
Comment (gsmet) [inline deployment/pom.xml:12]:
```suggestion
    <name>Quarkiverse - Tekton Client - Deployment</name>
```

File: deployment/pom.xml [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>io.quarkiverse.tekton.client</groupId>
```
Comment (gsmet) [inline deployment/pom.xml:5]:
```suggestion
        <groupId>io.quarkiverse.tektonclient</groupId>
```

File: deployment/pom.xml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        </path>
+                    </annotationProcessorPaths>
+                </configuration>
+            </plugin>
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemPropertyVariables>
+                        <java.util.logging.manager>org.jboss.logmanager.LogManager</java.util.logging.manager>
+                        <quarkus.log.level>DEBUG</quarkus.log.level>
```
Comment (gsmet) [inline deployment/pom.xml:59]: Maybe drop that?

File: deployment/src/main/java/io/quarkiverse/tekton/client/deployment/TektonClientProcessor.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
+package io.quarkiverse.tekton.client.deployment;
```
Comment (gsmet) [inline deployment/src/main/java/io/quarkiverse/tekton/client/deployment/TektonClientProcessor.java:1]:
```suggestion
package io.quarkiverse.tektonclient.deployment;
```
Sorry we have strict rules for package names and groupIds :)

File: deployment/src/main/java/io/quarkiverse/tekton/client/deployment/TektonClientProcessor.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.IndexDependencyBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+class TektonClientProcessor {
+
+    private static final String FEATURE = "quarkus-tekton-client";
```
Comment (gsmet) [inline deployment/src/main/java/io/quarkiverse/tekton/client/deployment/TektonClientProcessor.java:19]:
```suggestion
    private static final String FEATURE = "tekton-client";
```

File: deployment/src/main/java/io/quarkiverse/tekton/client/deployment/TektonClientProcessor.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+        LOG.debug("Registering Tekton classes for reflection");
+
+        IndexView index = combinedIndexBuildItem.getIndex();
+
+        String[] classes = index.getKnownClasses().stream()
+                .filter(classInfo -> classInfo.name().toString().startsWith("io.fabric8.tekton.pipeline"))
+                .map(Object::toString).toArray(String[]::new);
+
+        LOG.trace(Arrays.asList(classes));
```
Comment (gsmet) [inline deployment/src/main/java/io/quarkiverse/tekton/client/deployment/TektonClientProcessor.java:48]: I'm not sure I would keep the debug and traces now that you got it working but that's your call.

File: runtime/pom.xml [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+            <artifactId>tekton-client</artifactId>
+            <version>${tekton-client.version}</version>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.jboss.jandex</groupId>
+                <artifactId>jandex-maven-plugin</artifactId>
```
Comment (gsmet) [inline runtime/pom.xml:39]: Do you need that? Doesn't look useful?

**Review by gsmet — CHANGES_REQUESTED** [review]
I added some comments, mostly about our naming rules! Looks good otherwise.

File: runtime/src/main/java/io/quarkiverse/tekton/client/TektonClientProducer.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import io.fabric8.tekton.client.TektonClient;
+import io.quarkus.arc.DefaultBean;
+
+@Singleton
+public class TektonClientProducer {
+
+    @DefaultBean
+    @Singleton
+    @Produces
+    public TektonClient tektonClient(Config config) {
```
Comment (gsmet) [inline runtime/src/main/java/io/quarkiverse/tekton/client/TektonClientProducer.java:17]: Hmmm, another question, where is this `Config` object coming from?

**Comment by goldmann** [comment]
@gsmet I'll be taking care of these in a sec, thanks for the review!

File: runtime/src/main/java/io/quarkiverse/tekton/client/TektonClientProducer.java (hunk 9 above)
Comment (goldmann) [reply runtime/src/main/java/io/quarkiverse/tekton/client/TektonClientProducer.java:17]: It's from kubernetes-client extension: https://github.com/quarkusio/quarkus/blob/bce536fe09d3902aaee1a23cbca17939af98f9b0/extensions/kubernetes-client/runtime/src/main/java/io/quarkus/kubernetes/client/runtime/KubernetesClientProducer.java#L21 (which is a dependency for this extension).

File: runtime/pom.xml (hunk 8 above)
Comment (goldmann) [reply runtime/pom.xml]: I'm not exactly sure why I added it, maybe it was one of the things I tried when developing the extension and it was just left there. I removed it!

**Comment by goldmann** [comment]
@gsmet I've applied changes you requested and provided answer to your question. Ready for another round!

**Pull request merged**